src/mon/MgrMonitor.cc

```diff
@@ -854,8 +854,10 @@ bool MgrMonitor::preprocess_command(MonOpRequestRef op)
   }
 
   string format;
-  cmd_getval(g_ceph_context, cmdmap, "format", format, string("json-pretty"));
-  boost::scoped_ptr<Formatter> f(Formatter::create(format));
+  cmd_getval(g_ceph_context, cmdmap, "format", format,
+	     string("json-pretty"));
```

Comment:
i'd suggest pass an empty string to `cmd_getval()`. see https://github.com/ceph/ceph/pull/29089#discussion_r304467969 . as the `default_type`, which is the second parameter of `Formatter::create()` won't kick in if `format` is empty. but we need to use the `fallback` feature of `Formatter::create()`, so it can fallback to the `fallback` type if `format` is not supported.

it would be better to avoid the repeating. but if we have to, IMHO, it's better to repeat "json-pretty" in a single place.